forge/routes/ui/index.js

```diff
@@ -15,24 +15,42 @@
  */
 const path = require('path')
 const fs = require('fs')
+const fsp = require('fs').promises
 const Avatar = require('./avatar')
 
 module.exports = async function (app) {
     const frontendAssetsDir = path.join(__dirname, '../../../frontend/dist/')
 
+    /**
+     * Inject additional script imports for index.html that are dependant upon flowforge.yml
+     */
+    async function injectPlausible (domain, extension) {
+        const filepath = path.join(frontendAssetsDir, 'index.html')
+        const data = await fsp.readFile(filepath, 'utf8')
+        const injected = data.replace(/<!-- forge injected scripts go here -->/g,
+            `<script defer data-domain="${domain}" src="https://plausible.io/js/plausible${extension ? '.' + extension : ''}.js"></script>`)
+        return injected
+    }
+
```

Comment:
We only need to do the substitution once as the `flowforge.yml` config cannot change without a full restart. This proposed change caches the index.html in memory with the change applied.

```suggestion
    /**
     * Inject additional script imports for index.html that are dependant upon flowforge.yml
     */
    async function injectPlausible (domain, extension) {
        if (!cachedIndex) {
            const filepath = path.join(frontendAssetsDir, 'index.html')
            const data = await fsp.readFile(filepath, 'utf8')
            cachedIndex = data.replace(/<!-- forge injected scripts go here -->/g,
                `<script defer data-domain="${domain}" src="https://plausible.io/js/plausible${extension ? '.' + extension : ''}.js"></script>`)
        }
        return cachedIndex
    }

    let cachedIndex = null

```

Reply:
Should we host the plausible script ourselves rather than hardcode their CDN?

Reply:
That is a _plausible_ option